### PR TITLE
Added documentation for In-Stock Alerts (Magento 1 and 2)

### PR DESCRIPTION
**Pull request by Kagaminara**

### CODE REVIEW

File: source/docs/appendices/migration-guides.md [hunk 1]
```diff
@@ -9,11 +9,15 @@ Our goal is to make migrations as smooth as possible. This is why we try to make
 
 ## `2.14.0` -> `2.15.0`
 
-Nothing yet!
+### In-Stock Alerts
+
+In this release, we introduced the In-Stock Alert feature. When this feature is enabled on your backend (Magento 1 and 2), it will by default render a new component `SubscribeToInStockAlert` inside the existing `OutOfStock` component.
+
+Should you need to move the new component somewhere else, you will need to override `theme/modules/AddToCart/OutOfStock.js`.
```
Comment (mo-fc) [inline source/docs/appendices/migration-guides.md:16]: Not needed?

```suggestion
```

File: source/docs/advanced/features/in-stock-alert.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Customize In-Stock Alert texts
+
+By default, having the configuration enabled for in-stock alerts in your backend will automatically add the default component (`SubscribeToInStockAlert`) on each out-of-stock product page.
+
+The placeholders and messages displayed by the `SubscribeToInStockAlert` component have translation keys prefixed with `modules.SubscribeToInStockAlert`. You can customize the text from your application translations files.
+
+<div class="center">
+  <a class="link primary button" href="/docs/advanced/theme/translations.html">Learn about translations in Front-Commerce</a>
+</div>
```
Comment (mo-fc) [inline source/docs/advanced/features/in-stock-alert.md:36]: We do not usually mention this level of detail

File: source/docs/advanced/features/in-stock-alert.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+Navigate to `System > Configuration > Catalog > Catalog`. Within the `Product Alerts` menu, make sure `Allow Alert When Product Comes Back in-Stock` is set to `Yes`.
+
+### Magento 2
+
+Navigate to `Stores > Configuration > Catalog > Catalog`. Within the `Product Alerts` menu, make sure `Allow Alert When Product Comes Back in Stock` is set to `Yes`.
+
+## Customize In-Stock Alert texts
+
+By default, having the configuration enabled for in-stock alerts in your backend will automatically add the default component (`SubscribeToInStockAlert`) on each out-of-stock product page.
```
Comment (mo-fc) [inline source/docs/advanced/features/in-stock-alert.md:30]: Should they worry about caching? like is the configuration above cached in redis? if so then maybe mention it here

**Review by mo-fc — CHANGES_REQUESTED** [review]

File: source/docs/advanced/features/in-stock-alert.md (hunk 2 above)
Comment (Kagaminara) [reply source/docs/advanced/features/in-stock-alert.md:36]: I based what I wrote on [the quickorders' documentation](https://github.com/front-commerce/developers.front-commerce.com/blob/main/source/docs/advanced/features/quickorder.md), in which we do include such details. Did our guideline changed in the meantime?

File: source/docs/advanced/features/in-stock-alert.md (hunk 3 above)
Comment (Kagaminara) [reply source/docs/advanced/features/in-stock-alert.md:30]: That's a good question, I'm really not familiar with how FC's caching works yet, so I don't think I would be able to answer that. Would you mind helping me figure that one out?

**Comment by abailleulfc** [comment]
Don't forget the preview links 
* https://deploy-preview-389--heuristic-almeida-1a1f35.netlify.app/docs/advanced/features/overview.html
* https://deploy-preview-389--heuristic-almeida-1a1f35.netlify.app/docs/appendices/migration-guides.html
* https://deploy-preview-389--heuristic-almeida-1a1f35.netlify.app/docs/reference/environment-variables.html

File: source/docs/advanced/features/overview.md [hunk 4]
```diff
@@ -13,5 +13,6 @@ Front-Commerce has several extension points that allow to override default imple
 - [SmartForms](./smart-forms.html)
 - [Quick order](./quickorder.html)
 - [Password fields](./password-fields.html)
+- [In-stock alert](./in-stock-alert.html)
```
Comment (abailleulfc) [inline source/docs/advanced/features/overview.md:16]: Is it on purpose to not add it to the left menu ? 
image

File: source/docs/advanced/features/in-stock-alert.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+---
+id: in-stock-alert
+title: In-Stock Alert
+---
+
+<blockquote class="feature--new">
+_Since version 2.15_
+</blockquote>
+
+In-stock alerts allow users to subscribe to email notifications when a product are out of stock, so they can be notified as soon as it is back in stock.
```
Comment (abailleulfc) [inline source/docs/advanced/features/in-stock-alert.md:10]:
```suggestion
In-stock alerts allow users to subscribe to email notifications when a product is out of stock, so they can be notified as soon as it is back in stock.
```

File: source/docs/advanced/features/in-stock-alert.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+To use in-stock alerts in your project, you must first make sure the feature is enabled in your backend configuration.
+
+### Magento 1
+
+Navigate to `System > Configuration > Catalog > Catalog`. Within the `Product Alerts` menu, make sure `Allow Alert When Product Comes Back in-Stock` is set to `Yes`.
+
+### Magento 2
+
+Navigate to `Stores > Configuration > Catalog > Catalog`. Within the `Product Alerts` menu, make sure `Allow Alert When Product Comes Back in Stock` is set to `Yes`.
+
```
Comment (abailleulfc) [inline source/docs/advanced/features/in-stock-alert.md:27]: The action is the same
```suggestion
### Magento 1 & Magento 2

Navigate to `System > Configuration > Catalog > Catalog`. Within the `Product Alerts` menu, make sure `Allow Alert When Product Comes Back in-Stock` is set to `Yes`.

```

**Review by abailleulfc — CHANGES_REQUESTED** [review]

File: source/docs/advanced/features/overview.md (hunk 4 above)
Comment (Kagaminara) [reply source/docs/advanced/features/overview.md:16]: Nope, forgot about it :+1:

File: source/docs/advanced/features/in-stock-alert.md (hunk 6 above)
Comment (Kagaminara) [reply source/docs/advanced/features/in-stock-alert.md:27]: Magento 1's path is `**System** > Configuration` and Magento 2's is `**Stores** > Configuration` though ?

Comment (abailleulfc) [reply source/docs/advanced/features/in-stock-alert.md:27]: subbtle :open_mouth: forget about it then :innocent:

Comment (Kagaminara) [reply source/docs/advanced/features/in-stock-alert.md:27]: No worries ! 😁 It is subtle indeed !

File: source/docs/advanced/features/in-stock-alert.md (hunk 2 above)
Comment (mo-fc) [reply source/docs/advanced/features/in-stock-alert.md:36]: Not sure there is a guideline for those. I just didnt come across something like this before. But if you have seen it in another place then fair enough

**Review by mo-fc — APPROVED** [review]

**Review by abailleulfc — APPROVED** [review]

**Pull request merged**